plugins/in_calyptia_fleet/in_calyptia_fleet.c

```diff
@@ -0,0 +1,558 @@
+        flb_plg_error(ins, "http status code error: %d", client->resp.status);
+        goto http_error;
+    }
+
+    if (client->resp.payload_size <= 0) {
+        flb_plg_error(ins, "empty response");
+        goto http_error;
+    }
+
+    /* copy and NULL terminate the payload */
+    data = flb_sds_create_size(client->resp.payload_size + 1);
+    if (!data) {
+        goto http_error;
+    }
+    memcpy(data, client->resp.payload, client->resp.payload_size);
+    data[client->resp.payload_size] = '\0';
+
+    ret = header_lookup(client, "Last-Modified: ", strlen("Last-Modified: "), 
+                        &fbit_last_modified, &fbit_last_modified_len);
+    if (ret == -1) {
+        flb_plg_error(ctx->ins, "unable to get last-modified header");
+        goto http_error;
+    }
+
+    // Wed, 21 Oct 2015 07:28:00 GMT
+    flb_strptime(fbit_last_modified, "%a, %d %B %Y %H:%M:%S GMT", &tm_last_modified);
+    time_last_modified = mktime(&tm_last_modified.tm);
+
+    cfgname = time_fleet_config_filename(ctx, time_last_modified);
+    if (access(cfgname, F_OK) == -1 && errno == ENOENT) {
+        cfgfp = fopen(cfgname, "w+");
+        header = flb_sds_create_size(4096);
+        flb_sds_printf(&header, 
+                       "[INPUT]\n"
+                       "    Name          calyptia_fleet\n"
+                       "    Api_Key       %s\n"
+                       "    fleet_id      %s\n"
+                       "    Host          %s\n"
+                       "    Port          %d\n"
+                       "    TLS           %d\n",
+                       ctx->api_key,
+                       ctx->fleet_id,
+                       ctx->ins->host.name,
+                       ctx->ins->host.port,
+                       ctx->ins->tls_verify
+        );
+        fwrite(header, strlen(header), 1, cfgfp);
+        flb_sds_destroy(header);
+        fwrite(data, client->resp.payload_size, 1, cfgfp);
+        fclose(cfgfp);
+
+        cfgnewname = new_fleet_config_filename(ctx);
+        if (exists_new_fleet_config(ctx) == FLB_TRUE) {
+            cfgoldname = old_fleet_config_filename(ctx);
+            rename(cfgnewname, cfgoldname);
+            unlink(cfgnewname);
+            flb_sds_destroy(cfgoldname);
+        }
+        link(cfgname, cfgnewname);
+
+        // FORCE THE RELOAD!!!
+        flb_plg_info(ctx->ins, "force the reloading of the configuration file=%d.", ctx->event_fd);
+        flb_sds_destroy(cfgname);
+        flb_sds_destroy(data);
+
+        execute_reload(ctx, cfgnewname);
+        FLB_INPUT_RETURN(0);
+    }
+
+    flb_sds_destroy(cfgname);
+    flb_sds_destroy(data);
+    ret = 0;
+
+http_error:
+    flb_http_client_destroy(client);
+client_error:
+    flb_upstream_conn_release(u_conn);
+conn_error:
+    FLB_INPUT_RETURN(ret);
+}
+
+static void create_fleet_directory(struct flb_in_calyptia_fleet_config *ctx)
+{
+    flb_sds_t myfleetdir;
+
+    if (access("/tmp/calyptia-fleet", F_OK)) {
```

Comment:
nits: We don't need to consider for Windows for now. Although, `/tmp` does not exist on Windows.
Need to handle with some of environment variables or APIs later.

Reply:
Yeah I think we should include Windows and macOS ideally if we can: a variable would be fine.

Reply:
This has been pushed. I was able to test the windows portion some what by inverting the `#ifdef` and such. Hopefully it compiles at least.